dirs/dirs.go

```diff
@@ -204,28 +205,59 @@ func SnapHomeDirs() []string {
 	defer snapHomeDirsMu.Unlock()
 	dirs := make([]string, len(snapHomeDirs))
 	copy(dirs, snapHomeDirs)
+	// Should never be true since SetSnapHomeDirs is ran on init.
+	// Usefull for unit tests.
+	if len(dirs) == 0 {
+		return []string{filepath.Join(GlobalRootDir, "/home")}
+	}
 	return dirs
 }
 
-// SetSnapHomeDirs sets SnapHomeDirs to the user defined values and appends /home if needed
-// Homedir must be a comma separated list of the user defined home directories
-// If homedirs is empty, SnapHomeDirs will be a slice of length 1 containing "/home"
-func SetSnapHomeDirs(homedirs string) {
+// SetSnapHomeDirs sets SnapHomeDirs to the user defined values and appends /home if needed.
+// Homedir must be a comma separated list of the user defined home directories.
+// If homedirs is empty, SnapHomeDirs will be a slice of length 1 containing "/home".
+// Also generates the data directory globbing expressions for each user.
+// Ran at startup by configstate.Init to initialize snapHomeDirs, SnapDataHomeGlob and HiddenSnapDataHomeGlob.
+func SetSnapHomeDirs(homedirs string) []string {
 	snapHomeDirsMu.Lock()
 	defer snapHomeDirsMu.Unlock()
+	homeDirs = homedirs
 	snapHomeDirs = strings.Split(homedirs, ",")
 	for i := range snapHomeDirs {
-		// make sure the paths are clean, necessary for unit tests
+		if !strings.HasPrefix(snapHomeDirs[i], GlobalRootDir) {
```

Comment:
you might have to make a local copy of GlobalRootDir and add / if it's not at the end of it, otherwise this is not a proper check for directory prefix